frame/democracy/src/types.rs

```diff
@@ -0,0 +1,218 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Miscellaneous additional datatypes.
+
+use crate::{AccountVote, Conviction, Vote, VoteThreshold};
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{Bounded, CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, Saturating, Zero},
+	DispatchError, RuntimeDebug,
+};
+
+pub trait OriginMap<AssetId, Origin> {
+	fn try_origin_for(asset_id: AssetId) -> Result<Origin, DispatchError>;
+}
+
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct ProposalId<Hash, AssetId> {
```

Comment:
move ProposalId and OriginMap into other file. please make back this types generic out of hash. and this file will be near 100% as original